pkgs/applications/misc/openlp/lib.nix

```
@@ -0,0 +1,101 @@
+# This file contains the base package, some of which is compiled.
+# Runtime glue to optinal runtime dependencies is in 'default.nix'.
+{ fetchurl, stdenv, qt5
+
+# python deps
+, python, buildPythonPackage
+, alembic, beautifulsoup4, chardet, lxml, Mako, mock, nose, pyenchant
+, pyqt5_with_qtwebkit, pyxdg, sip, sqlalchemy, sqlalchemy_migrate
+}:
+
+buildPythonPackage rec {
+  pname = "openlp";
+  version = "2.4.6";
+
+  src = fetchurl {
+    url = "https://get.openlp.org/${version}/OpenLP-${version}.tar.gz";
+    sha256 = "f63dcf5f1f8a8199bf55e806b44066ad920d26c9cf67ae432eb8cdd1e761fc30";
+  };
+
+  doCheck = false;
+  # FIXME: checks must be disabled because they are lacking the qt env.
+  #        They fail like this, even if built and wrapped with all Qt and
+  #        runtime dependencies:
+  #
+  #     running install tests
+  #     qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
+  #     This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
+  #
+  #     Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx.
+  #
+  # See also https://discourse.nixos.org/t/qt-plugin-path-unset-in-test-phase/
+
+  #checkInputs = [ mock nose ];
+  nativeBuildInputs = [ qt5.qttools ];
```

```suggestion
  nativeBuildInputs = [ qttools ];
```

libsForQtX should not use qt5 as an input.

---

Why? This is in the python package and gets qt5 from libsForQt5, so it's synchronized. Though, I also include qttools as a dependency in the outer package without using it there.

I see mainly four main options:

1. Make everything one package. → rebuild necessary on every runtime dep change
2. Put `lib.nix` into the closure of the outer package in `default.nix`. → compromises clearly visible distinction between runtime and compile time stuff
3. Make `lib.nix` a function of `qttools`. → additional dependence between the two files – isn't this more complicated?
4. Leave as is.

---

This way you can mix qt versions.

> This is in the python package and gets qt5 from libsForQt5, so it's synchronized.

I am not fully sure this is the case. I would be more confident if you would use qttools directly.

---

I investigated: It is in principle synchronized since 2e9c639cb2a7ddf74956da2a945f8c9420f51297, but not in `pythonPackages.callPackage` because this is in another scope (see https://github.com/NixOS/nixpkgs/pull/102168#issuecomment-776090401).

But using `qt5` still doesn't add any problem: It is already there by using pyqt5. If pyqt5 (so pythonPackages) uses another qt version than the rest of the package you are totally screwed anyway. The only way to override qt consistently in a python+qt5 package is to override nixpkgs as described in the comment referenced above.